src/Sarif/Writers/SarifLogger.cs

```diff
@@ -243,7 +251,8 @@ public void AnalysisStarted()
 
         public void AnalysisStopped(RuntimeConditions runtimeConditions)
         {
-            if (_run.Invocations != null && _run.Invocations.Count > 0)
+            if (_run.Invocations != null && _run.Invocations.Count > 0 &&
+                !_dataToRemove.HasFlag(OptionallyEmittedData.NondeterministicProperties))
```

Comment:
>NondeterministicProperties [](start = 61, length = 26)

Why do we assign `EndTimeUtc` in two different places? And if we have to do it, why is the code that decides whether to assign it different in the two places? Doesn't 
```
if (_run?.Invocations?.Count > 0 && !_dataToRemove.HasFlag(OptionallyEmittedData.NondeterministicProperties))
```
work in both cases? And if it does work in both places, and this code needs to exist in both places, we should DRY it out into `ShouldEmitEndTime`.

(In fact, I often DRY out logic like that into a helper even if it only exists once, for documentation purposes.)